library/core/src/num/dec2flt/table.rs

```diff
@@ -1,1277 +1,670 @@
-//! Tables of approximations of powers of ten.
+//! Pre-computed tables powers-of-5 for extended-precision representations.
+//!
+//! These tables enable fast scaling of the significant digits
+//! of a float to the decimal exponent, with minimal rounding
+//! errors, in a 128 or 192-bit representation.
+//!
 //! DO NOT MODIFY: Generated by `src/etc/dec2flt_table.py`
 
-pub const MIN_E: i16 = -305;
-pub const MAX_E: i16 = 305;
+pub const SMALLEST_POWER_OF_FIVE: i32 = -342;
+pub const LARGEST_POWER_OF_FIVE: i32 = 308;
+pub const N_POWERS_OF_FIVE: usize = (LARGEST_POWER_OF_FIVE - SMALLEST_POWER_OF_FIVE + 1) as usize;
 
+// Use static to avoid long compile times: Rust compiler errors
+// can have the entire table compiled multiple times, and then
+// emit code multiple times, even if it's stripped out in
+// the final binary.
```

Comment:
Could we file a ticket with a repro case? I go to great lengths to avoid emitting multiple redundant errors and it'd be nice to improve things for our end users so they don't have to do these kind of things too :)

Reply:
Consts are by definition copy-pasted at every use of them. Statics by definition are not duplicated.

Reply:
Also note that what's copied here is data in the final binary, not errors.

Reply:
@estebank I'm not sure it requires an issue, but this is the [commit](https://github.com/rust-lang/rust/commit/bd51dea6934d5c583fe9c8bc806e6287cac43f90) and [PR](https://github.com/rust-lang/rust/pull/82676) that reduce the compilation time.

This seems to have been fixed, in a very large array that replicates the above behavior:

```rust
// Sample repository to check const vs static compile times.

pub const ARRAY: ([u64; 100000], [u64; 100000]) = ( 
    /* Two arrays of 100,000 elements, randomly generated */ 
);

#[inline(always)]
fn get_value(index: usize) -> (u64, u64) {
     let x = ARRAY.0[index];
     let y = ARRAY.1[index];
     (x, y)
}

macro_rules! array_vec {
    ($($i:expr),*$(,)*) => ({
        #[allow(unused_mut)] {
            let mut vec = Vec::new();
            $(vec.push(get_value($i));)*
            vec
        }
    });
}

fn mul128(x: (u64, u64), y: (u64, u64)) -> (u64, u64) {
    (x.0.overflowing_mul(y.0).0, x.1.overflowing_mul(y.1).0)
}

pub fn main() {
    let v = array_vec![... /* 400 random indexes in [0, 100000] */];
    let product = v.iter().fold((1, 1), |x, &y| mul128(x, y));
    println!("product is: {:?}", product);
}
```

```
# Static example
$ time rustc main.rs -C opt-level=3

real    0m2.967s
user    0m2.950s
sys     0m0.162s

# Const example
$ time rustc main.rs -C opt-level=3

real    0m3.033s
user    0m3.020s
sys     0m0.173s
```

Reply:
Note: this might still occur, so I see no downside to using `static`, but I'm not sure a compiler diagnostic is required if people are using new versions of Rustc.